File: genlearn-ai/backend/app/utils/languages.py
"""
Language constants for multi-language support.
Maps language codes to human-readable names.
"""

# Supported languages with their codes and names
SUPPORTED_LANGUAGES = {
    "en": "English",
    "zh": "Mandarin Chinese",
    "es": "Spanish",
    "hi": "Hindi",
    "pt": "Portuguese",
    "bn": "Bengali",
    "ru": "Russian",
    "ja": "Japanese",
    "pa": "Punjabi",
    "vi": "Vietnamese",
    "ar": "Arabic",
}

DEFAULT_LANGUAGE = "en"


def get_language_name(code: str) -> str:
    """Get the language name from a language code."""
    return SUPPORTED_LANGUAGES.get(code, "English")


def get_language_instruction(language: str) -> str:
    """
    Generate a strong language instruction for AI prompts.
    Ensures AI responds in the specified language.
    """
    if language == "en" or language not in SUPPORTED_LANGUAGES:
        return ""
    
    lang_name = SUPPORTED_LANGUAGES[language]
    return f"""
CRITICAL LANGUAGE REQUIREMENT: You MUST respond entirely in {lang_name}.
- Write ALL your responses in {lang_name} language
- Only keep technical terms, proper nouns, or brand names in their original form
- Do NOT respond in English unless specifically discussing English language concepts
- This instruction overrides all other language preferences
"""

File: genlearn-ai/frontend/src/constants/languages.ts
/**
 * Supported languages for multi-language chat feature.
 * Each language has a code, English name, and native name for display.
 */

export interface Language {
    code: string;
    name: string;
    nativeName: string;
}

export const SUPPORTED_LANGUAGES: Language[] = [
    { code: 'en', name: 'English', nativeName: 'English' },
    { code: 'zh', name: 'Mandarin Chinese', nativeName: '中文' },
    { code: 'es', name: 'Spanish', nativeName: 'Español' },
    { code: 'hi', name: 'Hindi', nativeName: 'हिन्दी' },
    { code: 'pt', name: 'Portuguese', nativeName: 'Português' },
    { code: 'bn', name: 'Bengali', nativeName: 'বাংলা' },
    { code: 'ru', name: 'Russian', nativeName: 'Русский' },
    { code: 'ja', name: 'Japanese', nativeName: '日本語' },
    { code: 'pa', name: 'Punjabi', nativeName: 'ਪੰਜਾਬੀ' },
    { code: 'vi', name: 'Vietnamese', nativeName: 'Tiếng Việt' },
    { code: 'ar', name: 'Arabic', nativeName: 'العربية' },
];

export const DEFAULT_LANGUAGE = 'en';

export function getLanguageName(code: string): string {
    const lang = SUPPORTED_LANGUAGES.find((l) => l.code === code);
    return lang ? lang.name : 'English';
}

export function getLanguageNativeName(code: string): string {
    const lang = SUPPORTED_LANGUAGES.find((l) => l.code === code);
    return lang ? lang.nativeName : 'English';
}
